### ISSUE TITLE
Distribution across multi-gpu nodes

### ISSUE THREAD

**Issue by SeanNaren** [kubeflow/pytorch-operator#128]
Thanks for the work in this! This is somewhat tied #30, but I'm used to using DistributedDataParallel with a script similar to [this](https://github.com/NVIDIA/apex/blob/master/apex/parallel/multiproc.py) to use multi-gpu for speed/performance over the DataParallel wrapper!

I've started using kubeflow for single GPU nodes, but I'm curious if there is any way I could use two separate 8GPU nodes to train while using the DistributedDataParallel locally on each 8GPU node? Anything I can do to help include this?

**Comment by SeanNaren** [comment]
@jwwandy seems like for now this is a good enough solution and is working with a modified version of the launch script! Once i've migrated changes to a public branch I'll close this ticket with a link to the fixes

**Comment by hyperparameters** [comment]
@jwwandy how do set ip address of master pod `--master_addr` this argument , when the job is created the master and worker pods are up at the same time.. so how do we know beforehand the ip of master pod.
All the other environment variable can be set, need some help with this
also do we have to launch our script with `python -m torch.distributed.launch`

**Comment by jwwandy** [comment]
@hyperparameters It has been a while since last time I worked on it. From my experience, Pytorch-operator will set up environment variable `MASTER_ADDR` for your worker pods. It should work fine if you collect environment variables and pass it into that pytorch script. However, Kubeflow has released numerous new versions in the past 18 months, so the information I provided might be outdated. Try it and let me know if there's anything I could help.

The code where the operator set up the env var is here: https://github.com/kubeflow/pytorch-operator/blob/61fefa88f75b126fd7672f44b87351db511299cb/pkg/controller.v1/pytorch/pod.go#L255
